src/snowflake/snowpark/dataframe.py

```diff
@@ -1095,48 +1093,31 @@ def drop(
         Args:
             *cols: the columns to exclude, as :class:`str`, :class:`Column` or a list
                 of those.
-
-        Raises:
-            :class:`SnowparkClientException`: if the resulting :class:`DataFrame`
-                contains no output columns.
         """
-        # An empty list of columns should be accepted as dropping nothing
+
         if not cols:
             raise ValueError("The input of drop() cannot be empty")
```

Comment:
cannot be empty or cannot be `None`?

Reply:
I moved the "An empty lost of columns.." comment a few lines down to the relevant piece of code. 

The `if not cols` check was present before in `drop()`, it prevents empty calls like `df.drop()`. `df.drop([])` is allowed though.